packages/react-select/src/components/LiveRegion.js

```diff
@@ -0,0 +1,167 @@
+// @flow
+/** @jsx jsx */
+import React, { type Node, useMemo } from 'react';
+import { jsx } from '@emotion/react';
+import A11yText from '../internal/A11yText';
+import {
+  defaultAriaLiveMessages,
+  type AriaSelectionType,
+} from '../accessibility';
+
+import type { CommonProps, OptionType, OptionsType, ValueType } from '../types';
+
+// ==============================
+// Root Container
+// ==============================
+
+export type LiveRegionProps = CommonProps & {
+  children: Node,
+  innerProps: { className?: string },
+  // Select state variables
+  ariaSelection: AriaSelectionType,
+  focusedOption: OptionType,
+  focusedValue: ValueType,
+  selectValue?: ValueType,
+  focusableOptions: OptionsType,
+  isFocused: boolean,
+};
+
+const LiveRegion = (props: LiveRegionProps) => {
+  const {
+    ariaSelection,
+    focusedOption,
+    focusedValue,
+    focusableOptions,
+    isFocused,
+    selectValue,
+    selectProps,
+  } = props;
+
+  const {
+    ariaLiveMessages,
+    getOptionLabel,
+    inputValue,
+    isMulti,
+    isOptionDisabled,
+    isSearchable,
+    menuIsOpen,
+    options,
+    screenReaderStatus,
+    tabSelectsValue,
+  } = selectProps;
+  const ariaLabel = selectProps['aria-label'];
+  const ariaLive = selectProps['aria-live'];
+
+  // Update aria live message configuration when prop changes
+  const messages = useMemo(
+    () => ({
+      ...defaultAriaLiveMessages,
+      ...(ariaLiveMessages || {}),
+    }),
+    [ariaLiveMessages]
+  );
+
+  // Update aria live selected option when prop changes
+  const ariaSelected = useMemo(() => {
+    let message = '';
+    if (ariaSelection && messages.onChange) {
+      const { value, ...context } = ariaSelection;
+      message = messages.onChange(value, { ...context });
+    }
+    return message;
+  }, [ariaSelection, messages]);
+
+  const ariaFocused = useMemo(() => {
+    let focusMsg = '';
+    const focused = focusedOption || focusedValue;
+    const isSelected = !!(
+      focusedOption &&
+      selectValue &&
+      selectValue.includes(focusedOption)
```

Comment:
this fails to determine `isSelected` if `isMulti` and multiple options are chosen.  I tried out react-fast-compare (which doesn't appear to be used anywhere) and made it work; for your consideration:

```suggestion
      !!selectValue.find((selected) => exportedEqual(selected, focusedOption))
```

that function imported at the top: 

```js
import exportedEqual from '../internal/react-fast-compare';
```

Reply:
Thanks, let me take a look at this and add some tests to make sure this doesn't get missed.

Reply:
Do you have latest and can you provide an example of it not working correctly? I do not seem to have the same problem of the select identifying menu options that are selected.
<img width="855" alt="Screen Shot 2021-03-01 at 3 51 14 PM" src="https://user-images.githubusercontent.com/1007579/109564125-3f4b6e80-7aa6-11eb-8fb7-17ac7269aef3.png">

Reply:
I do have latest.  I don't know how to provide a good  example, but I'll try.  I'm using hooks:

```tsx
const FRUITS_AND_VEGGIES = [
    'Apple',
    'Artichoke',
    'Asparagus',
    'Banana',
    'Beets',
    'Bell pepper',
    'Broccoli',
    'Brussels sprout',
]
export const MultipleValues = () => {
    const options = FRUITS_AND_VEGGIES.map((x, i) => ({
        value: x,
        label: x,
        isDisabled: i % 5 === 1,
    }));

    const [value, setValue] = useState<any>(options[2]);

    return (
        <div>
            <label id="ex3-label">Fruits and Vegetables</label>
            <Select
                isMulti
                value={value}
                onChange={(v) => setValue(v)}
                options={options}
                closeMenuOnSelect={false}
                aria-labelledby="ex3-label"
                isSearchable
                isClearable={false}
            />
       </div>
    );
};
```

Here's what I'm getting:

<img width="613" alt="Screen Shot 2021-03-01 at 6 31 45 PM" src="https://user-images.githubusercontent.com/1410985/109568952-08815280-7abe-11eb-893e-33f55e90d892.png">

But it's a different message initially:

<img width="609" alt="Screen Shot 2021-03-01 at 6 32 30 PM" src="https://user-images.githubusercontent.com/1410985/109569002-19ca5f00-7abe-11eb-8497-34328cfd606b.png">

Reply:
Oh, you know what's happening is that the function is being rerendered, and the array map is generating new objects (in my example).  So your test is checking for identical objects, which fails in my case.  If it were defined outside of the function component, then it should work with the simple equality check. Makes sense.  For your consideration.

Reply:
There is some internal discussion regarding the conversation of comparing objects by value or reference. I don't disagree with a value based approach over one that is referential, but opted to remain consistent with the existing approach for the sake of consistency.